org.eclipse.winery.repository/src/main/java/org/eclipse/winery/repository/backend/IGenericRepository.java

```diff
@@ -436,13 +437,15 @@
 
         final TNodeType nodeType = this.getElement(id);
 
-        // add all referenced requirement types
-        TNodeType.RequirementDefinitions reqDefsContainer = nodeType.getRequirementDefinitions();
-        if (reqDefsContainer != null) {
-            List<TRequirementDefinition> reqDefs = reqDefsContainer.getRequirementDefinition();
-            for (TRequirementDefinition reqDef : reqDefs) {
-                RequirementTypeId reqTypeId = new RequirementTypeId(reqDef.getRequirementType());
-                ids.add(reqTypeId);
+        // add all referenced requirement types, but only in XML mode. YAML does not have requirement types
+        if (!Environments.getUiConfig().getFeatures().get("yaml")) {
```

Comment:
Use constant? Isn't there a constant somewhere?

Reply:
Pls check the rest of the code as well ;-)

Reply:
Check `org.eclipse.winery.common.configuration.UiConfigurationObject`. If I wanted to use a constant, it would look like: `RepositoryConfigurationObject.RepositoryProvider.YAML.name().toLowerCase()`

Reply:
There is no overall enum, or a set of constants for features if that is what you are asking for..